.github/workflows/tox.yml

```diff
@@ -0,0 +1,145 @@
+name: tox
+
+on:
+  create:  # is used for publishing to TestPyPI
+    tags:  # any tag regardless of its name, no branches
+      - "**"
+  push:  # only publishes pushes to the main branch to TestPyPI
+    branches:  # any integration branch but not tag
+      - "master"
+  pull_request:
+  release:
+    types:
+      - published  # It seems that you can publish directly without creating
+  schedule:
+    - cron: 1 0 * * *  # Run daily at 0:01 UTC
+
+jobs:
+  build:
+    name: ${{ matrix.tox_env }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - tox_env: lint
+          - tox_env: docs
+          # - tox_env: py35
+          # - tox_env: py36
+          - tox_env: py37
+          # - tox_env: py38
+          # - tox_env: py39
+          - tox_env: packaging
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update \
+          && sudo apt-get install -y  ansible \
+          && ansible-doc -l | grep docker_container
+      # Login needed to increase rate-limits
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Find python version
+        id: py_ver
+        shell: python
+        if: ${{ contains(matrix.tox_env, 'py') }}
+        run: |
+          v = '${{ matrix.tox_env }}'.split('-')[0].lstrip('py')
+          print('::set-output name=version::{0}.{1}'.format(v[0],v[1:]))
+      # Even our lint and other envs need access to tox
+      - name: Install a default Python
+        uses: actions/setup-python@v2
+        if: ${{ ! contains(matrix.tox_env, 'py') }}
+      # Be sure to install the version of python needed by a specific test, if necessary
+      - name: Set up Python version
+        uses: actions/setup-python@v2
+        if: ${{ contains(matrix.tox_env, 'py') }}
```

Comment:
This is complex because it covers to testing all the other python versions which are now commented. I would keep it like this to avoid wasting time when we enable them.

Reply:
Ok, but our test suite take 11 minutes due to huge use of subprocess/docker build etc. I'm pretty sure this is not good for climate change :D  
What if we restrict other python version tests to test only tests not requiring the "host" fixture ? So only pure python tests, we have some of them and I think they could catch most errors on newer python versions (let say new keywords etc).